### ISSUE TITLE
TypeScriptBuilder vs TS Server Watch mode

### ISSUE THREAD

**Issue by boniatillo-com** [angelozerr/typescript.java#214]
What features are provided by TypeScriptBuilder that are not provided by the TS Server Watch mode?

Probably to use `tsserver -w` could be a simpler solution to issues like for [allowJs parameter ](https://github.com/angelozerr/typescript.java/issues/213).

**Comment by angelozerr** [comment]
TS Server Watch mode seems beeing a new feature and perhaps VScode uses that? We must study that https://github.com/Microsoft/vscode/tree/master/extensions/typescript/src 

Any PR are welcome!

**Comment by boniatillo-com** [comment]
Yes, we should look at that because I think the Watch mode could simplify a lot TypeScript IDE, probably an Eclipse IDE is not going to be needed anymore. I think the idea behind this kind of language services (like tsserver and lsp) is that clients don't need to do any heavy stuff, all the state and computation should be on the server side.

**Comment by angelozerr** [comment]
> Yes, we should look at that because I think the Watch mode could simplify a lot TypeScript IDE, probably an Eclipse project builder is not going to be needed anymore.

The main problem with that is that you cannot see progress monitor and stop as soon as possible the compile process.

**Comment by boniatillo-com** [comment]
But it think it is a problem of `tsserver`, not a problem of the IDE. The `tsserver` should provide a protocol for monitoring (compilation progress and cancelation).

**Comment by angelozerr** [comment]
>  The tsserver should provide a protocol for monitoring (compilation progress and cancelation).

Not sure that TypeScript teams will implement that.

**Comment by boniatillo-com** [comment]
I guess they should if they want to be IDE independent. In the meantime, to stop the compilation we can kill the process, and print the output messages to show the progress.

**Comment by angelozerr** [comment]
> In the meantime, to stop the compilation we can kill the process

No wa cannot do that, because tsserver is used too for completion, hover, etc

>  and print the output messages to show the progress.

Perhaps tsserver watch supports that. The best mean is to study this feature, but I have no time to study that for the moment.

Any help are welcome:)

**Comment by boniatillo-com** [comment]
Yes, the tsserver emits the files are being compiled.

I was looking into the VSCode TypeScript extension you pointed to me above and I did not find any related to the watch mode, however in the official tutorial about TypeScript development with VSCode they say the compiling is made via Tasks, traduced to the Eclipse world, via Run Configurations:

https://code.visualstudio.com/docs/languages/typescript#_transpiling-typescript-into-javascript

It means that the user is the one who executes the compiler, in watch mode or not. I think this is could be valid for TypeScript IDE, instead of implement a builder it can provides simpler ways of running the typescript compiler, for example, with Run Configurations or a TypeScript Compiler console.

A TypeScript Compiler console could be a kind of terminal that adds `tsc` to the exec path.

**Comment by angelozerr** [comment]
> with Run Configurations or a TypeScript Compiler console.

This launch already exists https://github.com/angelozerr/typescript.java/wiki/TypeScript-Compiler

You must not to use compileOnSave in this case.

**Comment by boniatillo-com** [comment]
I have to test it :), actually, I see you mention there the watch mode.

**Comment by angelozerr** [comment]
@boniatillo-com  I think when you say tsserver watch mode, it's tsc watch mode. It's not the same thing.

The benefit with tsserver compile on save is that you avoid double parsing of ts files (one by tsserver and one by tsc). More with tsserver you can manage progress monitor.

**Comment by boniatillo-com** [comment]
Hi @angelozerr, sorry, I am not collaborating more, I am preparing the papers myself for two conventions and my time is zero. As soon as I back from the travel I will return on this. I want to deliver the new major Phaser Editor version with typescript ide.
